pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,121 @@ package sdkstats
 }
 
-//Search returns Github API results for Code Search, for a given Query string
-func Search(client *github.Client, q string) ([]*github.CodeSearchResult, error) {
-
-	var searchOp []*github.CodeSearchResult
-	ctx := context.Background()
+// search returns Github API results for Code Search, for a given Query string
+func search(ctx context.Context, tc *http.Client, q string) ([]*github.CodeSearchResult, error) {
+	var searchop []*github.CodeSearchResult
+	client := github.NewClient(tc)
 
 	opts := &github.SearchOptions{TextMatch: true,
 		ListOptions: github.ListOptions{
 			PerPage: 100,
 		},
 	}
+	//Below Logic enables to collate Search results from all pages returned from API call.
 	for {
 		op, resp, err := client.Search.Code(ctx, q, opts)
 		if err != nil {
 			return nil, err
 		}
-		searchOp = append(searchOp, op)
+		searchop = append(searchop, op)
 		if resp.NextPage == 0 {
 			break
 		}
 		opts.Page = resp.NextPage
 	}
-	return searchOp, nil
-
+	return searchop, nil
 }
 
-//GetRepoDetails Returns map[string][string] with Stars,CreatedAt,PushedAt, and TotalCommits details for a given onwer, and repo
-func GetRepoDetails(client *github.Client, owner string, name string) map[string]string {
-
-	repoMap := map[string]string{}
+// getRepoDetails Returns []RepoMetaData with Stars,CreatedAt,PushedAt, and TotalCommits details for a given owner, and repo.
+func getRepoDetails(ctx context.Context, tc *http.Client, codeResults github.CodeResult, rq RepoMetadataQuery) (*RepoMetadata, error) {
+	var totalCommits int
+	client := github.NewClient(tc)
 
-	ctx := context.Background()
+	repoOwner := codeResults.GetRepository().GetOwner().GetLogin()
+	repoName := codeResults.GetRepository().GetName()
 
-	repos, _, err := client.Repositories.Get(ctx, owner, name)
+	version, err := rq.VersionParser.ParseVersion(codeResults)
 	if err != nil {
-		fmt.Println(err)
-	}
-	commits, _, err := client.Repositories.ListContributorsStats(ctx, owner, name)
-	var total int
-	for _, r := range commits {
-		total = total + r.GetTotal()
+		return nil, err
 	}
-	repoMap["Stars"] = strconv.Itoa(repos.GetStargazersCount())
-	repoMap["CreatedAt"] = repos.GetCreatedAt().String()
-	repoMap["PushedAt"] = repos.GetPushedAt().String()
-	repoMap["TotalCommits"] = strconv.Itoa(total)
-	return repoMap
-
-}
 
-func getversion(codeResults github.CodeResult, projectType string) int {
-
-	var searchQ string
-	var SDKversion int
-
-	if projectType == "helm" {
-		searchQ = "quay.io/operator-framework/helm-operator:v0."
+	repo, _, err := client.Repositories.Get(ctx, repoOwner, repoName)
+	if err != nil {
+		return nil, err
 	}
-	if projectType == "ansible" {
-		searchQ = "quay.io/operator-framework/ansible-operator:v0."
+	commits, _, err := client.Repositories.ListContributorsStats(ctx, repoOwner, repoName)
+	if err != nil {
+		return nil, err
 	}
-	if projectType == "go.mod" {
-		searchQ = "github.com/operator-framework/operator-sdk v0."
+	for _, r := range commits {
```

Comment:
To make it clearer, declare `totalCommits := 0` right above this loop rather than at the beginning of the function.

Reply:
done